express/scripts/preference-store.js

```diff
@@ -23,26 +23,72 @@
  * /express/blocks/feature-grid-desktop/feature-grid-desktop.js#L147
  */
 
+/** pre-defining all the accepted eventNames */
+export const preferenceNames = {
+  reduceMotion: {
+    name: 'reduceMotion',
+    mediaQueryString: '(prefers-reduced-motion: reduce)',
+  },
+};
+
 class PreferenceStore {
   constructor() {
     this.stores = {}; // { [name]: { value, subscribers: [HTMLElement] } }
   }
 
+  updateStore(name) {
+    const value = localStorage.getItem(name) === 'on';
+
+    if (!this.stores[name]) this.stores[name] = { subscribers: [] };
+    const store = this.stores[name];
+
+    store.value = value;
+    store.subscribers.forEach((sub) => {
+      sub.dispatchEvent(new CustomEvent(name, { detail: store }));
+    });
+    return store;
+  }
+
   /**
    * Sample usage:
    * import preferenceStore, { eventNames } from '../../scripts/preference-store.js';
-   * preferenceStore.set(eventNames.reduceMotion, sessionStorage.getItem('reduceMotion'));
+   * preferenceStore.init('reduceMotion');
    */
-  set(name, value) {
-    if (!this.stores[name]) this.stores[name] = { subscribers: [] };
-    const store = this.stores[name];
+  init(name) {
```

Comment:
maybe it's unnecessary to consider the case where the init() is called multiple times, but if that happens, duplicate callbacks could get added